### PR TITLE
reduce image size with multi-stage builds

### PR DESCRIPTION
**Pull request by tuanhtrng**
Got the image size down to 37MB from 184MB.

Tested on Debian Stretch, docker client 18.01.0-ce.

### CODE REVIEW

**Comment by JAremko** [comment]
@tuanhtrng  That's great. Have you tested it with DockerHub builds?

File: alpine-vim-base/Dockerfile [hunk 1]
```diff
@@ -13,31 +18,31 @@ RUN apk add --update --virtual build-deps \
...
-    --prefix /usr \
+    --prefix /tmp/build \
     --with-features=big \
     --with-python-config-dir=/usr/lib/python2.7/config \
-    && make install \
-    && apk del build-deps \
-    && apk add \
+ && make install
+ 
+ FROM alpin:latest
```
Comment (JAremko) [inline alpine-vim-base/Dockerfile]: Typo

**Comment by JAremko** [comment]
I tested multi stage build at DockerHub they seems to work fine.

File: alpine-vim-base/Dockerfile [hunk 2]
```diff
@@ -13,31 +18,31 @@ RUN apk add --update --virtual build-deps \
     ncurses-dev \
     python \
     python-dev \
```
Comment (JAremko) [inline alpine-vim-base/Dockerfile]: This `\` needs to be removed.

Comment (JAremko) [inline alpine-vim-base/Dockerfile]: This string should end with `\`

**Comment by JAremko** [comment]
@tuanhtrng So I tested it locally and tried to build build out of it. But now I get errors like `E484: Can't open file /tmp/build/share/vim/syntax/syntax.vim` so You'll need to reword your PR. You has to build Vim with the same prefix.

**Comment by tuanhtrng** [comment]
A fix has been tested locally and pushed.

**Comment by JAremko** [comment]
@tuanhtrng Seems good to me. I cherry picked it. Thanks  :+1:

**Pull request closed without merging**